yabt/buildcontext.py

```diff
@@ -456,19 +456,38 @@ def build(target: Target):
                 built_targets.add(target.name)
                 target.done()
 
-                # TODO: retry flaky tests N times
+                # TODO: retry flaky tests N times AFTER all the rest passed
                 # TODO: collect stats and print report at the end
                 # TODO: support both "fail fast" (exit on first failure) and
                 #       run all tests (don't exit on first failure) modes
                 target_tested = False
                 if run_tests and 'testable' in target.tags:
                     if self.conf.no_test_cache or not test_cached:
                         logger.info('Testing target {}', target.name)
-                        test_start = time()
-                        self.test_target(target)
+                        retries = (
+                            target.props.retries or self.conf.test_retries)
+                        if retries < 0:
+                            raise ValueError(
+                                'Retries value must be positive: got {}'
+                                .format(retries))
+                        test_start = 0
+                        target.summary['fail_count'] = 0
```

Comment:
IIUC, you save this in the `summary` only so you can use it in a test assertion, but IIRC everything you put in `summary` will also get to the target disk cache, so if you don't need it in the cache (and don't want to invalidate all the caches) - better don't put it there

Reply:
Oh, so where's the correct place to accumulate stats and outputs?
In this PR I only have fail_count but we'll add logs in the future and a timing per re-run maybe

Reply:
if it's not relevant for caching - just for tracking stuff for a single run - then add some fields to the Target data structure itself (or just one field that is itself a nested data structure for tracking retries or something)